src/lib/yang/schema.lua

```diff
@@ -700,7 +707,10 @@ function resolve(schema, features)
                   node.body[k] = v
                end
                -- TODO: Handle refine and augment statements.
-            else
+            end
+         end
+         for k,v in pairs(node.body or {}) do
```

Comment:
So. I added inheriting the "config" property. I.e. a configuration option on a parent such as a container will be copied over to the leaf unless the leaf itself defines a config attribute. This is so I can easily check if any leaf has a config attribute without having to pass the current config attribute. It simplifies code which handles it.

In the simple example of:
```yang
module fruit {
      namespace "urn:testing:fruit";
      prefix "fruit";

      import ietf-inet-types {prefix inet; }
      import ietf-yang-types {prefix yang; }

      organization "Fruit Inc.";
      contact "John Smith fake@person.tld";
      description "Module to test YANG schema lib";

      revision 2016-05-27 {
         description "Revision 1";
         reference "tbc";
      }

      grouping fruit {
         description "Represents a piece of fruit";

         leaf name {
            type string;
            mandatory true;
            description "Name of fruit.";
         }
         leaf score {
            type uint8 {
               range 0..10;
            }
            mandatory true;
            description "How nice is it out of 10";
         }
         leaf tree-grown {
            type boolean;
            description "Is it grown on a tree?";
         }
      }

      container fruit-bowl {
         description "Represents a fruit bowl";
         list contents {
            config true;
            uses fruit;
         }
      }
   }
```

On the iteration when it's got to the `contents` list the first element is a "uses" statement (scm.kind == "uses"). The code goes ahead and looks up the grouping, removes the uses statement and then adds the leaves "name", "score" and "tree-grown" to the body. The iteration continues and then it iterates over "score" and "tree-grown" and now copies the "config true" which is now in `env.env.config` but not for "name". This is because the first iteration it didn't iterate over "name" it iterated over the uses statement, it's just name took it's place.

I added an addition for loop to iterate over to catch this quirk. It ensures when there is a grouping statement where `uses` is expanded, all the children of the expanded `uses` still inherit config correctly if it's there to inherit.